securedrop_client/gui/widgets.py

```diff
@@ -839,33 +839,31 @@ def update(self, sources: List[Source]):
         """
         Reset and update the list with the passed in list of sources.
         """
-        # A flag to show if a source is currently selected (the current source
-        # doesn't have to be selected in a QListWidget -- it appears to default
-        # to whatever is in row 0, whether it's selected or not).
-        has_source_selected = False
-        if self.currentRow() > -1:
-            has_source_selected = self.item(self.currentRow()).isSelected()
-        current_source = self.get_current_source()
-        current_source_id = current_source and current_source.id
-
-        # When we call clear() to delete all SourceWidgets, we should
-        # also clear the source_widgets dict.
-        self.clear()
-        self.source_widgets = {}
-
+        # Delete widgets that no longer exist in source list
+        source_uuids = [source.uuid for source in sources]
+        for i in range(self.count()):
+            list_item = self.item(i)
+            list_widget = self.itemWidget(list_item)
+
+            if list_widget and list_widget.source.uuid not in source_uuids:
```

Comment:
if a source has been deleted, I think we'll get a `ObjectDeletedError` from accessing `source.uuid` (you can trigger this scenario for yourself by deleting a source via the JI)

Reply:
that makes sense - any time we access a db object in the gui we should put it in a try-catch

Reply:
looks like we use source.uuid all over the code, so we will need a followup pr to try-catch for the object deletion error that you mention